06-building-a-program-start-to-finish/writing-tests/Output/test/spec/test.js

```diff
@@ -0,0 +1,83 @@
+const chai = require('chai');
+const expect = chai.expect;
+ 
+ const InvertedIndex = require('../../src/js/inverted-index.js');
+ const books = require('../books.json');
+ 
+ const emptyBook = [];
+ const newIndex = new InvertedIndex();
+ const sampleData = ' Mary had a    little #$%^6 lamb, a little lamb Mary had';
+ newIndex.createIndex('books.json', books);
+ 
+ describe('Validate files', () => {
+   it('should check that uploaded file is valid JSON', () => {
+     expect(Array.isArray(books)).to.be.true;
+     expect(Array.isArray(emptyBook)).to.be.true;
+     expect(typeof books[0]).to.equal('object');
+   });
+ 
+   it('should not be empty', () => {
+     expect(books.length).to.not.equal(0);
+     expect(emptyBook.length).to.equal(0);
+   });
+   it("should have a valid 'title' and 'text'", () => {
+     expect(books[0].title).to.not.be.undefined;
+     expect(books[0].text).to.not.be.undefined;
+   });
+ });
+ 
+ describe('Normalized Text', () => {
+   it('should return an array containing only alphabets', () => {
+     expect(newIndex.normalizedText(sampleData)).to.not.include(' ');
+     expect(newIndex.normalizedText(sampleData)).to.not.include('#$%^6');
```

Comment:
Is this exhaustive? You might want to use `RegEx` to test that the returned value only contains alphabets.